### ISSUE TITLE
Ctrl-T with no selection does not stage all files anymore

### ISSUE THREAD

**Issue by clone** [git-for-windows/git#572]
In Git GUI 2.6.3.Windows.1, pressing Ctrl-T without files selected in Unstaged Changes, does not commit all files. That behaviour was present in previous versions. It is a very useful shortcut, as often you want to commit all files at once. It is probably related that in this version always a single line is selected by default. So probably something changed in the selection mechanism.

**Comment by dscho** [comment]
> That behaviour was present in previous versions.

Please understand that this handwaving statement puts a *ton* of a burden on anybody who is willing to help you resolve your issue. This is not fair. You should extend the courtesy of being as helpful as possible to anybody who is willing to help you.

**Comment by dscho** [comment]
> without files selected in Unstaged Changes

Also: why do you make it so hard to follow your report by not telling anybody how to have *no file* selected? In my hands, there is always one file selected.

Please understand that this makes it pretty frustrating to even try to help you.

**Comment by clone** [comment]
I'm sorry about the version. I noticed the behaviour only after installing the newer version. I cannot reliably determine which version the previously installed version was. I think it was 1.9.5, but I'm not sure.

I also noticed that in this version always ONE file was selected, while previously it by default none was selected, and you had to click a file. If you did not click, and then pressed CTRL-T, ALL files were staged, which was a VERY useful shortcut.

I'm not sure if I can reinstall an older version, but I will give it a shot. If I find that version, I will update here. On the other hand, I can imagine that the behaviour of selection of files and pressing CTRL-T is probably limited to one or two source files, so looking at the history of those should reveal where a change was made.

**Comment by dscho** [comment]
> I also noticed that in this version always ONE file was selected, while previously it by default none was selected, and you had to click a file.

That still leaves out any information how to make sure that no file is selected. Your original report mentioned that *if* no file is selected, it does not work, but it left out information *how* no file can be selected.

In any case, with a more complete bug report, you might have more luck on the [Git mailing list](mailto:git@vger.kernel.org), as this issue seems to be independent of the OS.

**Comment by patthoyts** [comment]
In git-gui, `Control-T` is bound to `do_add_selection` which requires something to be selected. `Control-I`  is bound to `do_add_all`. Hitting Ctrl-I when new files are available to be staged pops up a dialog asking if you want to stage the untracked files and will stage all other changes for you.

I don't think Control-T ever worked the way you claim. I checked git-gui-0.16 up to 0.20 (current) and none of these work as claimed. I suspect Control-I is what you remember using.

**Comment by dscho** [comment]
@patthoyts thanks for clarifying!

**Comment by clone** [comment]
Well, the good news is that CTRL-I works to add all files.

Maybe the CTRL-T behaviour I described was a bug with a useful side-effect which was somehow changed in later versions, because after some trying I am able to force CTRL-T to stage ALL files while NONE where shown to be selected in Git GUI 2.7.0.Windows.1.

Steps to reproduce:
1. Have working copy with 1 changed file, 2 untracked files.
2. Start Git GUI
3. CTRL-I, confirm staging untracked files
4. Select ALL files
5. CTRL-U
Now all files are shown UNSELECTED
6. CTRL-T will now stage ALL files, CTRL-U will unstage ALL files

It could be a display issue, hence I then did
7. Rescan
This shows that one file is selected
8. CTRL-T will still stage ALL files, CTRL-U will unstage ALL files, and no files are shown to be selected.

At this point CTRL-T "works" as I described as previously seen behaviour.
So my suspicion is a change in the selection behaviour causes this.

Maybe previously when no file was selected, by default all files were selected?

**Comment by clone** [comment]
To add to this: the CTRL-T behaviour as describes with no files selected is actually MORE useful than CTRL-I, because it does NOT ask for confirmation of adding untracked files. So I would still like to see this behaviour restored.

**Comment by dscho** [comment]
> So I would still like to see this behaviour restored.

@clone as @patthoyts said:

> I don't think Control-T ever worked the way you claim.

I do not think that there is anything to be restored. If you want to introduce this feature, why don't you give it a whirl?
